tests/test_custom_learner.py

```diff
@@ -229,3 +239,54 @@ def test_custom_learner_api_missing_file():
 def test_custom_learner_api_bad_extension():
     other_dir = join(_my_dir, 'other')
     _ = Learner('_CustomLogisticRegressionWrapper', custom_learner_path=join(other_dir, 'custom_learner.txt'))
+
+
+@raises(ValueError)
+def test_custom_learner_learning_curve_min_examples():
+    """
+    Test to check learning curve raises error with less than 500 examples
+    """
+    # generates a training split with less than 500 examples
+    train_fs_less_than_500, _ = make_classification_data(num_examples=499,
+                                                         train_test_ratio=1.0,
+                                                         num_labels=3)
+
+    # creating an example learner
+    learner = Learner('LogisticRegression')
+
+    # this must throw an error because `examples` has less than 500 items
+    _ = learner.learning_curve(examples=train_fs_less_than_500, metric='accuracy')
+
+
+def test_custom_learner_learning_curve_min_examples_override():
+    """
+    Test to check learning curve displays warning with less than 500 examples
+    """
+
+    # creates a logger which writes to a temporary log file
+    log_dir = join(_my_dir, 'log')
+    log_file = NamedTemporaryFile("w", delete=False, dir=log_dir)
+    logger = get_skll_logger("test_custom_learner_learning_curve_min_examples_override",
+                             filepath=log_file.name)
+
+    # generates a training split with less than 500 examples
+    train_fs_less_than_500, _ = make_classification_data(num_examples=499,
+                                                         train_test_ratio=1.0,
+                                                         num_labels=3)
+
+    # creating an example learner
+    learner = Learner('LogisticRegression', logger=logger)
+
+    # this must throw an error because `examples` has less than 500 items
+    _ = learner.learning_curve(examples=train_fs_less_than_500, metric='accuracy',
+                               override_minimum=True)
+
+    # checks that the learning_curve warning message is contained in the log file
+    with open(log_file.name) as tf:
+        log_text = tf.read()
+        learning_curve_warning_re = \
+            re.compile(r'Because the number of training examples provided - '
+                       r'\d+ - is less than the ideal minimum - \d+ - '
+                       r'learning curve generation is unreliable'
+                       r' and might break')
+        assert learning_curve_warning_re.search(log_text)
```

Comment:
These look great @srhrshr ! One minor thing: is there a reason why these test are in `test_custom_learner.py` rather than `test_output.py` where the other learning curve tests are? These don't have anything to do with custom learners right?

Reply:
That's a good point - I didn't realize there were learning curve tests as I was looking for tests of the `Learner*` class all this while 🤦   Will move the tests to `test_output.py`

Reply:
Done @desilinguist !